TrafficReplayer/src/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -59,32 +49,51 @@ public TrafficReplayer(URI serverUri)
     }
 
     private static void exitWithUsageAndCode(int statusCode) {
-        System.err.println("Usage: <log directory> <destination_url>");
+        System.err.println("Usage: <destination_url> [log file]");
         System.exit(statusCode);
     }
 
+    public static CloseableStringStreamWrapper getLogLineStreamFromInputStream(InputStream is) throws IOException {
+        InputStreamReader isr = new InputStreamReader(is);
+        try {
+            BufferedReader br = new BufferedReader(isr);
+            return CloseableStringStreamWrapper.generateStreamFromBufferedReader(br);
+        } catch (Exception e) {
+            isr.close();
+            throw e;
+        }
+    }
+
+
+    public static CloseableStringStreamWrapper getLogLineStreamFromFile(String filename) throws IOException {
+        FileInputStream fis = new FileInputStream(filename);
+        try {
+            return getLogLineStreamFromInputStream(fis);
+        } catch (Exception e) {
+            fis.close();
+            throw e;
+        }
+    }
+
     public static void main(String[] args) throws IOException, InterruptedException {
-        if (args.length != 2) {
+        if (args.length < 1 || args.length > 2) {
             exitWithUsageAndCode(1);
         }
 
         URI uri;
         try {
-            uri = new URI(args[1]);
+            uri = new URI(args[0]);
         } catch (Exception e) {
             exitWithUsageAndCode(2);
             return;
         }
-        Path directoryPath;
-        try {
-            directoryPath = Paths.get(args[0]);
-        } catch (Exception e) {
-            exitWithUsageAndCode(3);
-            return;
-        }
+
+        CloseableStringStreamWrapper css = args.length > 1 ? getLogLineStreamFromFile(args[1]) :
+                getLogLineStreamFromInputStream(System.in);
```

Comment:
I'm okay with this, but the convention that I'm used to is that `-`in the place of a file means to use stdin. That seems slightly more explicit and harder to mess up. Any reason you didn't go that route?

Reply:
Other than gunzip, where have you seen that convention?  It looks like cat & grep will assume stdin if nothing is specified, BUT will also interpret '-' to be stdin.  It's a fine change to make, but I feel like the arg parsing in general will need to be eventually improved anyway here.

Reply:
I think it's actually quite widespread.
https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html (second to last bullet point) is one source, "Many tools accept a bare hyphen, not associated with any option letter, as a pseudo-filename directing the application to read from standard input" [[source](http://www.catb.org/~esr/writings/taoup/html/ch10s05.html)] is a second, and [many non-authoritative stack overflow/github/google results](https://www.google.com/search?q=unix+convention+hyphen+as+stdin).

Totally fine with it being part of an argparsing improvement down the line though.